packages/core/package-manager/src/installPackage.js

```diff
@@ -131,18 +132,28 @@ async function determinePackageInstaller(
   filepath: FilePath,
 ): Promise<PackageInstaller> {
   let configFile = await resolveConfig(fs, filepath, [
-    'yarn.lock',
     'package-lock.json',
+    'pnpm-lock.yaml',
+    'yarn.lock',
   ]);
-  let hasYarn = await Yarn.exists();
 
-  // If Yarn isn't available, or there is a package-lock.json file, use npm.
   let configName = configFile && path.basename(configFile);
-  if (!hasYarn || configName === 'package-lock.json') {
+
+  // Always use the package manager that seems to be used in the project,
+  // falling back to a different one wouldn't update the existing lockfile.
+  if (configName === 'package-lock.json') {
     return new Npm();
+  } else if (configName === 'pnpm-lock.yaml') {
+    return new Pnpm();
+  } else if (configName === 'yarn.lock') {
+    return new Yarn();
   }
 
-  return new Yarn();
+  if (await Yarn.exists()) {
+    return new Yarn();
+  } else {
```

Comment:
I would say add a check if pnpm exists here too. If you have it installed I'd guess you want to use it, whereas npm comes with node...